kotlinpoet/src/main/java/com/squareup/kotlinpoet/TypeSpec.kt

```diff
@@ -340,7 +340,8 @@ public class TypeSpec private constructor(
       val property = propertySpecs[propertyIndex]
       if (property.getter != null || property.setter != null) continue
       val parameter = primaryConstructor.parameter(property.name) ?: continue
-      if (parameter.type != property.type) continue
+      // compare TypeName using `toString()` to account for type aliases
```

Comment:
Is there a failing test case for this condition? Feels a bit off, as a `TypeAliasSpec` would still point to the real `TypeName` through its `type` property, which I guess is what the `ParameterSpec` here is referencing?

Reply:
Thanks, the failing test is:
https://github.com/square/kotlinpoet/blob/master/interop/kotlinx-metadata/src/test/kotlin/com/squareup/kotlinpoet/metadata/specs/KotlinPoetMetadataSpecsTest.kt#L200

I spent more time looking into the underlying cause of the comparison failure, and it is actually because the tagMap are not equal because values of `TypeAliasTag` are not equal. I tried to fix this in 0619ef2b8

Reply:
Looks good, thanks!